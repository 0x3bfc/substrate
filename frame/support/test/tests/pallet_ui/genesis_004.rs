#[frame_support::pallet]
mod pallet {
	use frame_support::pallet_prelude::{ModuleInterface, GenesisBuild};
	use frame_system::pallet_prelude::BlockNumberFor;

	#[pallet::config]
	pub trait Trait: frame_system::Trait {}

	#[pallet::module]
	pub struct Module<T>(core::marker::PhantomData<T>);

	#[pallet::module_interface]
	impl<T: Trait> ModuleInterface<BlockNumberFor<T>> for Module<T> {}

	#[pallet::call]
	impl<T: Trait> Module<T> {}

	#[pallet::genesis_config]
	pub struct GenesisConfig;

	#[pallet::genesis_build]
	impl<T: Trait> GenesisBuild<T> for GenesisConfig {}
}

fn main() {
}
