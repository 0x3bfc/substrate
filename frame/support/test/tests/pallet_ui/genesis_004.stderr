error[E0277]: the trait bound `pallet::GenesisConfig: std::default::Default` is not satisfied
    --> $DIR/genesis_004.rs:22:18
     |
22   |     impl<T: Config> GenesisBuild<T> for GenesisConfig {}
     |                     ^^^^^^^^^^^^^^^ the trait `std::default::Default` is not implemented for `pallet::GenesisConfig`
     |
    ::: $WORKSPACE/frame/support/src/traits.rs
     |
     | pub trait GenesisBuild<T, I=()>: Default + MaybeSerializeDeserialize {
     |                                  ------- required by this bound in `frame_support::traits::GenesisBuild`
